### PR TITLE
DCA-457: Add a collaborative editing status check

### PR DESCRIPTION
**Pull request by mmizin**
https://bulldog.internal.atlassian.com/browse/DCA-457

### CODE REVIEW

File: app/util/data_preparation/api/confluence_clients.py [hunk 1]
```diff
@@ -139,6 +139,11 @@ def get_total_pages_count(self):
         response = self.get(api_url, 'Could not get issues count')
         return response.json().get('totalSize', 0)
 
+    def get_collaborative_editing_status(self):
+        api_url = f'{self.host}/rest/synchrony-interop/status'
+        response = self.get(api_url, error_msg='Could not get collaborative editing status').json()
+        return response
+
```
Comment (SergeyMoroz0703) [inline app/util/data_preparation/api/confluence_clients.py:146]: Good catch of `status` endpoint! But I figured out that there is option 'Limited' in Collaborative mode. In this case `Only one person will be able to edit at a time.` (doc). 
What do you think about add force enabling of collaborative editing?
`/rest/synchrony-interop/enable` this endpoint helps in this case, it starts a job for enabling CE(1-2sec).

Comment (mmizin) [reply app/util/data_preparation/api/confluence_clients.py:146]: Thx that you mentioned about the "Collaborative editing LIMITED"  option. We have to check the "synchronyEnabled" status too! I will add this check obligatorily.
As regards to force enabling of collaborative editing it is a good idea, that's what I thought too! I think I need to spend some time to investigate would it have a bad impact on us/vendors!

File: app/util/data_preparation/api/confluence_clients.py [hunk 2]
```diff
@@ -139,6 +139,11 @@ def get_total_pages_count(self):
         response = self.get(api_url, 'Could not get issues count')
         return response.json().get('totalSize', 0)
 
+    def get_collaborative_editing_status(self):
+        api_url = f'{self.host}/rest/synchrony-interop/status'
+        response = self.get(api_url, error_msg='Could not get collaborative editing status').json()
```
Comment (ometelytsia) [inline app/util/data_preparation/api/confluence_clients.py:144]: It's better not to mix response and response serialization.
```
response = self.get(api_url, error_msg='Could not get collaborative editing status')
return response.json
```

File: app/util/data_preparation/confluence/prepare-data.py [hunk 3]
```diff
@@ -90,6 +90,13 @@ def write_test_data_to_files(dataset):
     __write_to_file(CONFLUENCE_USERS, users)
 
 
+def __check_collaborative_editing_enable(confluence_api):
```
Comment (ometelytsia) [inline app/util/data_preparation/confluence/prepare-data.py:93]: For consistency let's rename method name to `__is_collaborative_editing_enabled`

File: app/util/data_preparation/confluence/prepare-data.py [hunk 4]
```diff
@@ -90,6 +90,13 @@ def write_test_data_to_files(dataset):
...
+    status = confluence_api.get_collaborative_editing_status()
+    if not all(status.values()):
+        raise Exception('Synchrony plugin and/or the collaborative editing feature are disabled in Confluence. '
+                        'Please turn collaborative editing back on.')
```
Comment (ometelytsia) [inline app/util/data_preparation/confluence/prepare-data.py:97]: Better wording here would be:
`Please turn on collaborative editing in Confluence System Preferences page in order to run DC Apps Performance Toolkit.`

File: app/util/data_preparation/api/confluence_clients.py (hunk 2 above)
Comment (mmizin) [reply app/util/data_preparation/api/confluence_clients.py]: Done

File: app/util/data_preparation/confluence/prepare-data.py (hunk 3 above)
Comment (mmizin) [reply app/util/data_preparation/confluence/prepare-data.py]: Done

File: app/util/data_preparation/confluence/prepare-data.py (hunk 4 above)
Comment (mmizin) [reply app/util/data_preparation/confluence/prepare-data.py]: Done

**Pull request merged**